test/test_images/sendevents/main.go

```diff
@@ -164,13 +165,15 @@ func main() {
 			}
 		}
 
-		if err := event.SetData(untyped); err != nil {
+		if err := event.SetData(cloudevents.ApplicationJSON, untyped); err != nil {
```

Comment:
Why don't you just set `[]byte(eventData)` more than unmarshalling and then marshal back?

Reply:
I am not reworking the code here, I am just updating to sdk v2